config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +45,93 @@
         this.bundleDocumentBuilder = bundleDocumentBuilder;
     }
 
-    public Element build(Bundle bundle, EntityBuilder.BundleType bundleType, Document document) {
+    public Map<String, Element> build(Bundle bundle, EntityBuilder.BundleType bundleType, Document document, String bundleName, String bundleVersion) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
 
-        return bundleDocumentBuilder.build(document, entities);
+        Map<String, Entity> annotatedEntities = new HashMap<>();
+        // Filter the bundle to export only annotated entities
+        // TODO : Enhance this logic to support services and policies
+        final Map<String, Encass> encassEntities = bundle.getEntities(Encass.class);
+        encassEntities.entrySet().parallelStream().forEach(encassEntry -> {
+            Encass encass = encassEntry.getValue();
+            if (encass.getAnnotations() != null) {
+                encass.getAnnotations().forEach(entityAnnotation -> {
+                    if (entityAnnotation.getType().equals(ANNOTATION_TYPE_BUNDLE)) {
+                        final String annotatedEntityName = entityAnnotation.getName();
+                        final String annotatedEntityDesc = entityAnnotation.getDescription();
+                        final Entity filteredEntity = entities.parallelStream().filter(entity -> encassEntry.getKey().equals(entity.getName())).findAny().get();
+                        annotatedEntities.put(annotatedEntityName != null ? annotatedEntityName + '-' + bundleVersion : encassEntry.getKey() + '-' + bundleVersion, filteredEntity);
```

Comment:
key is not just name it includes path as well

Reply:
encassEntry.getKey() returns name with path

Reply:
name with path like policy entity? I have tested this, it is returning only name